docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,669 @@
+        ".NETPortable,Version=v4.0,Profile=Profile328":{
+            "compilationOptions": {
+                "define": [ "PORTABLE328" ]
+            },
+            "frameworkAssemblies":{
+                "mscorlib":"",
+                "System":"",
+                "System.Core":"",
+                "System.Net"
+            }
+        }
+    }
+}
+
+```
+
+Now you can conditionally compile against that target:
+
+```csharp
+#if !PORTABLE328
+using System.Net.Http;
+using System.Threading.Tasks;
+// Potentially other namespaces which aren't compatible with Profile 328
+#endif
+```
+
+And that's it! Because `PORTABLE328` is now recognized by the compiler, the PCL Profile 328 library generated by a compiler will not include `System.Net.Http` or `System.Threading.Tasks`.
+
+## How to use multiple projects
+
+A common need for larger libraries is to place functionality in different projects.  For example, a library supplying functionality to both C# and F# consumers may have a core project, a C# project, and an F# project.  Here's how a potential folder structure would look:
+
+```
+/MyLibrary
+|__global.json
+|__/MyLibrary.Core
+   |__Sources
+   |__project.json
+|__/MyLibrary.CSharp
+   |__Sources
+   |__project.json
+|__/MyLibrary.FSharp
+   |__Sources
+   |__project.json
+```
+
+The `global.json` file would look like this:
+
+```javascript
+{
+    "projects":[
+        "MyLibrary.Core",
+        "MyLibrary.CSharp",
+        "MyLibrary.FSharp"
+    ]
+}
+```
+
+And the `project.json` for both `MyLibrary.FSharp` and `MyLibrary.CSharp` would include a reference to `MyLibrary.Core`:
+
+```javascript
+{
+    "dependencies":{
+        "MyLibrary.Core":""
+        ...
+    }
+    ...
+}
+```
+
+To build it all, you would open a command line at the root of the project (`/MyLibrary`).
+
+Using DNX:
+
+```
+$ dnu restore
+$ cd ../MyLibrary.CSharp
+$ dnu build
+$ cd ../MyLibrary.FSharp
+$ dnu build
+```
+    
+Using .NET CLI:
+   
+```
+$ dotnet restore
+$ cd ../MyLibrary.CSharp
+$ dotnet build
+$ cd ../MyLibrary.FSharp
+$ dotnet build
+```
+
+Note that `MyLibrary.Core` is automatically built because it was listed as a dependency.
+
+Any source files which use functionality in `MyLibrary.Core` can now just use it:
+
+C#:
+
+```csharp
+using MyLibrary.Core;
+```
+F#:
+
+```fsharp
+open MyLibrary.Core
+```
+
+And that's it!
+
+Important takeaways:
+
+* Projects are just folders with files in them.
+* There is a `global.json` file at the top level of your library which needs to names of each project (folder name).
+* A project using another project needs to register a dependency in its `project.json` before it can be used.
+
+## How to test libraries on .NET Core
+
+**Note: You can only use DNX to test at this time**
+
+It's important to be able to test across platforms.  You can do this using [Xunit](http://xunit.github.io/), which is also the testing tool used by .NET Core projects.  Getting a test to work across platforms can be done like this:
```

Comment:
At the moment I'm inclined to say no at the moment for NUnit because it's still not quite there yet.  As of right now, you need to build the test project and execute the binary it produces to run your tests.  IIRC, MSTest doesn't support .NET Core right now.  It looks like it has its own separate CLI experience for Windows, too.  Xunit feels "natural" to DNX because of the DNX command it added (and I'm sure will support with `dotnet test` in the future).